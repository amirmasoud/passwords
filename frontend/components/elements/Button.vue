<template>
  <button
    :type="type"
    :class="
      `flex items-stretch py-1 uppercase tracking-wide text-${color}-500 text-xs font-bold appearance-none border-none bg-${color}-700 text-${color}-200 rounded leading-tight`
    "
  >
    <div :class="`flex items-center bg-${color}-800 rounded shadow-lg mx-1`">
      <plus-svg :class="`text-${color}-200`" />
    </div>
    <div :class="`flex items-center text-${color}-100 pl-1 pr-2`">
      <slot></slot>
    </div>
  </button>
</template>

<script>
import PlusSvg from '~/components/svg/Plus'

export default {
  props: {
    type: {
      type: String,
      default: 'button'
    },
    color: {
      type: String,
      default: 'green'
    }
  },

  components: { PlusSvg }
}
</script>
