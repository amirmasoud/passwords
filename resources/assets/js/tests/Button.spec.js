import { mount } from '@vue/test-utils'
import Button from '../components/Button'

describe('Button', () => {
  const wrapper = mount(Button, {
    propsData: {
      type: 'secondary',
      nativeType: 'button',
      loading: true,
      block: true,
      large: true
    }
  })

  test('renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.setProps({ loading: false })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has a button', () => {
    expect(wrapper.contains('button')).toBe(true)
  })
})
