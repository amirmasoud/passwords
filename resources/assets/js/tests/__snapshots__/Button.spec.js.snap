// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `<button type="button" disabled="disabled" class="btn btn-block btn-lg btn-loading btn-secondary"></button>`;

exports[`Button renders correctly 2`] = `<button type="button" class="btn btn-block btn-lg btn-secondary"></button>`;
